### ISSUE TITLE
FEATURE: Support skips

### ISSUE THREAD

**Issue by wakatara** [blinry/habitctl#11]
Sometimes there are very good reasons to not do a task (sick, was out of town that week etc etc). It would be great if you would support skips ( y/n/s/- ?). That would simply out the task on hold for whatever period without affecting the graph or score in some way.

**Comment by blinry** [comment]
Good idea! I'm also often find myself being asked to do "impossible" habits – when you're out of town, but you're asked to clean your apartment, for example. So far, I've answered with "yes" in that case. But a skip feature would be great! I'll look into it, and I'm open to PRs! :)

**Comment by wakatara** [comment]
Awesome. Yeah I often find that's the case and want to skip it that particular alar week (though do notice a lot of travelling ends up being the main enemy I have to good habit formation.). 

(also, I'd totally love to help with PRs but I don't know Rust at all - as a learning exercise, I'm actually trying to duplicate your feature set in Go Lang :-) )

**Comment by wakatara** [comment]
I was thinking about this some more. Weirdly, I think you just need some sort of once character marker in the length of duration for the skipping (maybe that's what the - is for in the y/n/- though I kinda would like that to be a "don't answer right now" when filling in your daily stuff (I tend to update the habitctl as I go but because this is not supported end up editing the text file with vim quite frequently.).

For me, if you weren't already using it, I would have used the "." character you use currently for new day (and no record) as the skip indicator, but I guess a ✘ also would work for those cases though for cases where you have long weeks or like it's gonna look more like a "did not do" visually which feels wrong. So, still liking possibly repurposing the "." kinda like the way I think about ending a sentence on an ellipses... =] The other alternative would be a simple long dash (which would be above the underlines) but I think that would be confusing visually. 

What do you think? Of course the calculation for busyness would then need to "remove" that from the denominator count as well since, well... you're actively skipping those things.

**Comment by wakatara** [comment]
Hey there!

Trying to teach myself Rust, so I've built the feature I requested (😝 ). Pull request for this and a slight correction to the scoring calculation made for merging when you get a chance. I did have to change the default indicator for unknown from a * to a ?, but I also updaed the README and a few other cleanups. 

Lemme know how you feel about it. I've also pushed it to my fork in master over on wakatara. 

Again, thanks for writing this app. I know it sounds silly, but it seriously has helped me diagnose things and helped me attain goals that would have been harder without it (example: I am currently down 6kg and back in shape, have read 26 books this year already etc etc).
